compiler/src/iree/compiler/Codegen/Common/MaterializeTuningSpecsPass.cpp

```diff
@@ -197,14 +197,30 @@ struct MaterializeTuningSpecsPass final
       return;
     }
 
-    // If only the default tuning spec is available, use it directly and skip
-    // the linking stage.
-    if (!hasUserTuningSpec) {
-      if (failed(dumpFinalTuningSpecToDir(*defaultTuningSpec))) {
+    // Check if the user-provided tuning spec has the default entry point
+    // attribute.
```

Comment:
The code below is self-explanatory. I think the comment a few lines below should be enough.
```suggestion
```